Frends.Odoo.Request/Frends.Odoo.Request.Tests/UnitTests.cs

```diff
@@ -0,0 +1,123 @@
+namespace Frends.Odoo.Request.Tests;
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Definitions;
+using NUnit.Framework;
+
+[TestFixture]
+internal class UnitTests
+{
+    private readonly string odooUrl = Environment.GetEnvironmentVariable("ODOO_URL");
+    private readonly string username = Environment.GetEnvironmentVariable("ODOO_USERNAME");
+    private readonly string password = Environment.GetEnvironmentVariable("ODOO_PASSWORD");
+    private readonly string database = Environment.GetEnvironmentVariable("ODOO_DATABASE");
+
+    [Test]
+    public async Task Request_InvalidCredentials_Error()
+    {
+        // Arrange
+        var input = GetDefaultInput();
+        var options = GetDefaultOptions();
+        options.Password = "invalid";
+
+        // Act
+        var result = await Odoo.Request(input, options, CancellationToken.None);
+
+        // Assert
+        Assert.IsFalse(result.Success);
+        Assert.IsTrue(result.Error.StartsWith("Authentication failed. Odoo's response: Access Denied"), result.Error);
+        Assert.IsNull(result.Data);
+    }
+
+    [Test]
+    public async Task Request_InvalidDatabase_Error()
+    {
+        // Arrange
+        var input = GetDefaultInput();
+        var options = GetDefaultOptions();
+        options.Database = "invalid";
+
+        // Act
+        var result = await Odoo.Request(input, options, CancellationToken.None);
+
+        // Assert
+        Assert.IsFalse(result.Success);
+        Assert.IsTrue(result.Error.Contains("database \"invalid\" does not exist"), result.Error);
+        Assert.IsNull(result.Data);
+    }
+
+    [Test]
+    public async Task Request_InvalidCredentials_ThrowsErrorAsException()
```

Comment:
switch from "async Task" to "void" since with asert.throwAsync it's not async test anymore